pkg/model/deployment_test.go

```diff
@@ -78,5 +90,6 @@ func TestOverrideBackstageImage(t *testing.T) {
 	assert.NoError(t, err)
 
 	assert.Equal(t, "dummy", model.backstageDeployment.container().Image)
+	assert.Equal(t, "dummy", model.backstageDeployment.podSpec().InitContainers[0].Image)
```

Comment:
This test doesn't cover the case where the images for additional user-provided containers should not be modified. This, to me, was the whole point of RHIDP-1701 (issue with the `oauth2-proxy` container), and I think it was tested in https://github.com/janus-idp/operator/pull/279

Reply:
fixed